pkg/k8s/identitybackend/identity.go

```diff
@@ -100,11 +101,14 @@ func (c *crdBackend) AllocateID(ctx context.Context, id idpool.ID, key allocator
 		SecurityLabels: key.GetAsMap(),
 	}
 
-	_, err := c.Client.CiliumV2().CiliumIdentities().Create(ctx, identity, metav1.CreateOptions{})
-	return err
+	ci, err := c.Client.CiliumV2().CiliumIdentities().Create(ctx, identity, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return key.PutBackendKey(ci), nil
```

Comment:
Instead of storing the entire CiliumIdentities object, it should be sufficient to have a Created flag (or something similar) only. If the identity has just been created, then it is expected that it should not have been marked for deletion by the operator (the annotation would not be present in that case). 

Optionally, if useful, PutBackendKey could also be something like PutMeta (of WithValue), which allows to store arbitrary key/value pairs like in the case of contexts.